activerecord/lib/arel/tree_manager.rb

```diff
@@ -52,8 +52,9 @@ def to_dot
 
     def to_sql(engine = Table.engine)
       collector = Arel::Collectors::SQLString.new
-      collector = engine.lease_connection.visitor.accept @ast, collector
-      collector.value
+      engine.with_connection do |connection|
+        engine.lease_connection.visitor.accept(@ast, collector).value
```

Comment:
👋 I don't have much context on this PR, but based on the other changes, was this supposed to be

```suggestion
        connection.visitor.accept(@ast, collector).value
```

?

Reply:
Yes.